### PR TITLE
Removing local mode

### PR DESCRIPTION
**Pull request by ahmadnazeri**
# Description
Removing local mode from the client and cli so users cannot utilize local mode anymore. This does not remove all the code related to local mode but rather just access to it. 

### Select type(s) of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

# Checklist:

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have fixed any merge conflicts

### CODE REVIEW

File: client/src/featureform/cli.py [hunk 1]
```diff
@@ -157,36 +146,20 @@ def list(host, cert, insecure, local, resource_type):
...
+def version():
     client_version = get_package_version()
     host = os.getenv("FEATUREFORM_HOST", "")
     cluster_version = ""
     output = f"Client Version: {client_version}"
-    if local == False:
-        try:
-            cluster_version = get_version_hosted(host)
-        except:
-            cluster_version = "Cannot retrieve: Check your FEATUREFORM_HOST value. If using local mode, use the --local flag."
```
Comment (anthonylasso) [inline client/src/featureform/cli.py:170]: nice to get rid of this. always felt this was a clumsy solution.

File: client/tests/test_ondemand_features.py [hunk 2]
```diff
@@ -44,156 +44,156 @@ def test_fn():
...
-    ff.register_training_set(
-        "fraud_training",
-        "quickstart",
-        label=("fraudulent", "quickstart"),
-        features=[("avg_transactions", "quickstart")],
-    )
-
-    resource_client = ff.ResourceClient(local=True)
-    resource_client.apply()
+# @pytest.mark.local
```
Comment (anthonylasso) [inline client/tests/test_ondemand_features.py:47]: keeping this? or also swapping to hosted mode. if so I'd drop a comment

**Review by sdreyer — APPROVED** [review]

**Comment by dvdr00t** [comment]
Hello everyone, 
I was wondering why the local mode has been "removed" from the main package and if there is a valuable alternative to that. I have been using the PyPI package of Featureform version 1.11.2 and I need the provider to be a local provider as I have multiple `.csv` files from which I need to infere features. How does that work now?

Thanks!

**Comment by sdreyer** [comment]
Hey @dvdr00t this slipped past. For now localmode is effectively deprecated but can still be used in the older package versions. We're working on making it simpler to run the core codebase as a single executable locally, which will have access to local files and can act as a replacement to localmode for now. This is mainly due to the overhead of managing two separate code bases for the same functionality. The changes to the core backend should be complete in a couple weeks.

**Comment by dvdr00t** [comment]
Cool, many thanks for your reply, @sdreyer! Looking forward to the new version! 👍

**Comment by fumoboy007** [comment]
Hi! Is there documentation on how to access local files using the new approach?

**Pull request merged**